dipy/align/_public.py

```diff
@@ -0,0 +1,833 @@
+"""
+
+Registration API: simplified API for registration of MRI data and of
+streamlines.
+
+
+"""
+import collections
+import numbers
+import numpy as np
+import nibabel as nib
+from dipy.align.metrics import CCMetric, EMMetric, SSDMetric
+from dipy.align.imwarp import (SymmetricDiffeomorphicRegistration,
+                               DiffeomorphicMap)
+
+from dipy.align.imaffine import (transform_centers_of_mass,
+                                 AffineMap,
+                                 MutualInformationMetric,
+                                 AffineRegistration)
+
+from dipy.align.transforms import (TranslationTransform3D,
+                                   RigidTransform3D,
+                                   AffineTransform3D)
+
+import dipy.core.gradients as dpg
+import dipy.data as dpd
+from dipy.align.streamlinear import StreamlineLinearRegistration
+from dipy.tracking.streamline import set_number_of_points
+from dipy.tracking.utils import transform_tracking_output
+from dipy.io.streamline import load_trk
+from dipy.io.utils import read_img_arr_or_path
+
+
+# Global dicts for choosing metrics for registration:
+syn_metric_dict = {'CC': CCMetric,
+                   'EM': EMMetric,
+                   'SSD': SSDMetric}
+
+affine_metric_dict = {'MI': MutualInformationMetric}
+
+
+def _handle_pipeline_inputs(moving, static, static_affine=None,
+                            moving_affine=None, starting_affine=None):
+    """
+    Helper function to prepare inputs for pipeline functions
+
+    Parameters
+    ----------
+    moving, static: Either as a 3D/4D array or as a nifti image object, or as
+        a string containing the full path to a nifti file.
+
+    static_affine, moving_affine: 4D arrays
+
+    starting_affine : in case this is needed.
+    """
+    static, static_affine = read_img_arr_or_path(static,
+                                                      affine=static_affine)
+    moving, moving_affine = read_img_arr_or_path(moving,
+                                                      affine=moving_affine)
+    if starting_affine is None:
+        starting_affine = np.eye(4)
```

Comment:
`starting_affine = starting_affine or np.eye(4)` : The `if` is  not needed and this expression is always evaluated

Reply:
But sometimes `starting_affine` has a value that we would like to pass on as output.

Reply:
Oh  gotcha - you just want a oneliner here. OK!

Reply:
Done in 173b2d25c

Reply:
Looks like this doesn't work with arrays. See: https://travis-ci.org/github/dipy/dipy/jobs/685933762#L4454